packages/apollo-server/src/utils/types.ts

```diff
@@ -0,0 +1,99 @@
+import { GraphQLSchema } from 'graphql';
+import { CorsOptions } from 'cors';
+import { SchemaDirectiveVisitor, IResolvers } from 'graphql-tools';
+import { ConnectionContext } from 'subscriptions-transport-ws';
+import { GraphQLOptions } from 'apollo-server-core';
+export { CacheControlExtensionOptions } from 'apollo-cache-control';
+
+export type Context<T = any> = T;
+export type ContextFunction<T = any> = (
+  context: Context<T>,
+) => Promise<Context<T>>;
+
+export interface SubscriptionServerOptions {
+  path?: string;
+  onConnect?: Function;
+  onDisconnect?: Function;
+  keepAlive?: number;
+}
+
+export interface Config<Server, ContextShape = any, Cors = CorsOptions>
+  extends Pick<
+      GraphQLOptions<Context<ContextShape>>,
+      | 'formatError'
+      | 'debug'
+      | 'rootValue'
+      | 'logFunction'
+      | 'formatParams'
+      | 'validationRules'
+      | 'formatResponse'
+      | 'fieldResolver'
+      | 'debug'
+      | 'cacheControl'
+      | 'tracing'
+    > {
+  app?: Server;
+  typeDefs?: string | [string];
+  resolvers?: IResolvers;
+  schema?: GraphQLSchema;
+  schemaDirectives: Record<string, typeof SchemaDirectiveVisitor>;
+  context?: Context<ContextShape> | ContextFunction<ContextShape>;
+  cors?: Cors;
+  subscriptions?: SubscriptionServerOptions | string | false;
+  engineInRequestPath?: boolean;
+  engine?: boolean | Object;
+  enableIntrospection?: boolean;
```

Comment:
Would `introspection` be unreasonable here?  Having `engine` accept a boolean, but using the `enable` prefix for this seems slightly incongruent (though I realize they can be different).  I wonder if a future pattern could introduce passing an `Object` for the introspection, and removing `enable` could help support that with less friction.